internal/utils/flags/flags_test.go

```diff
@@ -0,0 +1,177 @@
+			expectedValues: []string{value1, value2, valueWithComma},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			values := []string{}
+			enumSetValue := NewEnumSetValue(&values, []string{}, validValues)
+			assert.Nil(t, enumSetValue.Set(tc.inputValue))
+			assert.Equal(t, tc.expectedValues, values)
+		})
+	}
+	t.Run("Invalid values should cause an error", func(t *testing.T) {
+		values := []string{}
+		enumSetValue := NewEnumSetValue(&values, []string{}, validValues)
+		expectedError := fmt.Errorf("unsupported value, use one of [%s] instead", strings.Join(validValues, ", "))
+		assert.Equal(t, expectedError, enumSetValue.Set("eggcorn"))
+	})
+}
+
+func TestEnumSetValueType(t *testing.T) {
+	t.Run("EnumSetValue should have type of stringSlice", func(t *testing.T) {
+		values := []string{}
+		enumSetValue := NewEnumSetValue(&values, []string{}, validValues)
+		assert.Equal(t, "stringSlice", enumSetValue.Type())
+	})
+}
+
+func TestEnumSetValueString(t *testing.T) {
+	for _, tc := range []struct {
+		description    string
+		values         []string
+		expectedString string
+	}{
+		{
+			description:    "No values should yield only brackets",
+			values:         []string{},
+			expectedString: "[]",
+		},
+		{
+			description:    "Values should be comma seperated",
+			values:         []string{value1, value2},
+			expectedString: fmt.Sprintf("[%s,%s]", value1, value2),
+		},
+		{
+			description:    "Values with commas should be printed within quotes",
+			values:         []string{valueWithComma},
+			expectedString: fmt.Sprintf(`["%s"]`, valueWithComma),
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			enumSetValue := NewEnumSetValue(&tc.values, tc.values, tc.values)
+			assert.Equal(t, tc.expectedString, enumSetValue.String())
+		})
+	}
+}
+
+func TestEnumSetValueAppend(t *testing.T) {
+	for _, tc := range []struct {
+		description    string
+		initialValues  []string
+		newValue       string
+		expectedValues []string
+	}{
+		{
+			description:    "A Valid value should be added",
+			initialValues:  []string{},
+			newValue:       value1,
+			expectedValues: []string{value1},
+		},
+		{
+			description:    "A duplicate value should not be added",
+			initialValues:  []string{value1},
+			newValue:       value1,
+			expectedValues: []string{value1},
+		},
+	} {
+		t.Run(tc.description, func(t *testing.T) {
+			values := []string{}
+			enumSetValue := NewEnumSetValue(&values, tc.initialValues, validValues)
+			assert.Nil(t, enumSetValue.(pflag.SliceValue).Append(tc.newValue))
+			assert.Equal(t, tc.expectedValues, values)
+		})
+	}
+	t.Run("Invalid values should cause an error", func(t *testing.T) {
+		values := []string{}
+		enumSetValue := NewEnumSetValue(&values, []string{}, validValues)
+		expectedError := fmt.Errorf("unsupported value, use one of [%s] instead", strings.Join(validValues, ", "))
+		assert.Equal(t, expectedError, enumSetValue.(pflag.SliceValue).Append("eggcorn"))
```

Comment:
these casts can go away in the `Replace`/`Append`/`GetSlice` tests if you change the return type of `NewEnumSet` to be `*EnumSetValue`